core/textile/model/mirror_file.go

```diff
@@ -89,11 +90,15 @@ func (m *model) CreateMirrorFile(ctx context.Context, mirrorFile *domain.MirrorF
 		return nil, err
 	}
 
-	_, err = m.FindMirrorFileByPathAndBucketSlug(ctx, mirrorFile.Path, mirrorFile.BucketSlug)
+	mf, err := m.FindMirrorFileByPathAndBucketSlug(ctx, mirrorFile.Path, mirrorFile.BucketSlug)
 	if err != nil {
 		return nil, err
 	}
 
+	if mf != nil {
+		return nil, errMirrorFileAlreadyExists
```

Comment:
@jsonsivar 

I wonder: should we return `err` or just `return mf, nil`?

Reply:
Will leave this as-is for now and can update later since it's working at least.  If we need a create or get type usage if this later we can always update.